Main.java

```diff
@@ -5,61 +5,58 @@
 
 public class Main {
 
-    public static void main(String[] args){
+    public static void main(String[] args) {
 
-        if(args.length!=2)
-        {
+        if (args.length != 2) {
             System.out.println("File path and search word cannot be empty");
-            System.exit(0);
+            return;
         }
-        String path=args[0];
-        String search=args[1];
-        File file=new File(path);
+        String path = args[0];
+        String search = args[1];
+        File file = new File(path);
         System.out.println("Processing...");
-        doesFileExist(file);
-        wordCount(file,search);
+        validateFile(file);
+        if (!validateFile(file)) {
+            wordCount(file, search);
+        }
     }
 
-    private static void doesFileExist(File file)
-    {
-        if (!file.isFile())
-        {
-            System.out.println("File does not exist");
-        } else isFileFormatSupported(file);
-    }
+    private static boolean validateFile(File file) {
 
-    private static void isFileFormatSupported(File file)
-    {
-        if (!file.getName().endsWith("txt"))
-        {
-            System.out.println("File format not supported");
+        if (!file.exists()) {
+            System.out.println("File does not exist");
+            System.exit(0);
+        } else {
+            String fullName = file.getName();
+            if (!(fullName.endsWith(".txt") || fullName.endsWith(".json"))) {

```

Comment:
What if used passes an extension other than .txt and .json?

Reply:
It will show "File format not supported"

Reply:
So it does not need the explicit check for .json right?

Reply:
The code needs to support both .txt and .json